text/0000-natvis.md

```diff
@@ -376,10 +407,77 @@ developers which may lead to a period of learning the syntax. Since this feature
+
+Supporting this option would mean that changes to cargo and rustc are not necessary.
+Each individual crate would be able to create a custom build script that would set
+the rustc `link-arg` flag `cargo:rustc-link-arg=/NATVIS:{file-path}` for each Natvis
+file.
+
+The drawbacks for this option is that it would force all Rust developers to manually
+create a build script and ensure it is kept up-to-date whenever the set of Natvis files
+are updated. This also would set the linker argument regardless of the linker being used.
+This could lead to potential build errors if the linker being used returns an error for
+an invalid argument.
+
+### Alternative 3: inline Natvis XML fragments via attributes only
+
+Supporting this option would mean that changes to cargo are not necessary.
+This option could be implemented via an attribute and/or proc-macro which
+would live outside of the compiler and could be ingested in via an outside crate.
+Rustc would need some changes in order to collect all of the attribute usage from the
+source code and create temporary files that could be passed to the MSVC linker via
+the `/NATVIS` linker arg. For crate dependencies, the Natvis fragments can be combined
+and embedded in the crate metadata so the Natvis can still be embedded in the final
+PDB generated.
+
+The drawbacks for this option is that it would add a lot of bloat to the Rust source
+code directly if only the attribute syntax was supported. For types with many fields
+or types that need extensive amounts of Natvis to appropriately visualize them in a
+meaninngful way, this could distract from the contents of the code. Without being able
+to pull some of the more intricate Natvis descriptions into a separate standalone
+`.natvis` file, there may become an issue with the visibility of the source code.
+Also, if/when other debugger visualization formats are supported, it could become
+very obscure to read the source with large amounts of visualization scripts from
+multiple schemas all being directly embedded in source code.
+
+### Alternative 4: miri executes the MIR of a Debug impl within a debugger
+
+Supporting this option would mean that changes to cargo and rustc are not necessary.
+This would have the added benefit of taking full advantage of existing implementations
+of the `Debug` trait. Many Rust developers already implement the `Debug` trait which is
+used to format how types should be viewed, this would only ease the debugging quality of
+Rust when viewed under any debugger. This option also has the added benefit of not
+requiring any changes to a crate from a Rust developer by leveraging existing `Debug` impls.
+
+The drawbacks for this option is that this has not been fully investigated to
+determine its viability. This could be a great potential feature to ease
+debugging Rust but without concrete data to push this towards a potential RFC,
+I would assume supporting debugging in the systems that are already heavily used
```

Comment:
I agree, but i am worried we'll settle for "ok instead of great" and never put any work into this without the pressure of improving debug visualizations

Reply:
I think this is always something of a concern when planning features. The feature this RFC is trying to implement is pretty well-defined and has clear value. The miri approach has yet to be investigated and while it may a great long-term solution, I think it needs to be investigated to ensure it can solve the problems mentioned above. I don't know if we want to block this feature on such an investigation into the miri solution.

Reply:
Yea... not sure who has the time to put in the work for that XD which is kind of my point. If we block it in that investigation, everyone who wants natvis has motivation to do the investigation.

But... I would probably feel annoyed if someone did that do me, so I wouldn't feel good to use that as leverage.